### ISSUE TITLE
gdal reports scale factor=90 for Polar Stereographic GRIB

### ISSUE THREAD

**Issue by teknocreator** [OSGeo/gdal#1064]
## Expected behavior and actual behavior.

When performing gdalinfo on Alaska Polar Stereographic GRIB files from the US National Weather Service National Digital Forecast Database (NDFD), I had expected the scale factor to be equal to 1 as with Mercator GRIB files.  However, gdal reports the scale factor to be equal to 90 instead for Alaska grids:

Driver: GRIB/GRIdded Binary (.grb)
Files: ndfdakgb3p018103018qpf06f012
Size is 1649, 1105
Coordinate System is:
PROJCS["unnamed",
    GEOGCS["Coordinate System imported from GRIB file",
        DATUM["unknown",
            SPHEROID["Sphere",6371200,0]],
        PRIMEM["Greenwich",0],
        UNIT["degree",0.0174532925199433]],
    PROJECTION["Polar_Stereographic"],
    PARAMETER["latitude_of_origin",60],
    PARAMETER["central_meridian",210],
    PARAMETER["scale_factor",90],
    PARAMETER["false_easting",0],
    PARAMETER["false_northing",0]]

So, when using this scale factor in a proj4 string in Openlayers or Mapserver such as 

+proj=stere +lat_0=90 +lat_ts=60 +lon_0=-150 +k=90 +x_0=0 +y_0=0 +a=6371200 +b=6371200 +units=m +no_defs

results in a VERY small image of Alaska for whatever weather element is encoded in GRIB.  Setting +k=1 corrects this issue.  Discovered this using QGIS, as I had to zoom up quite a bit when scale factor is reported as 90 and results in a normal-sized image when setting the same proj4 string with +k=1.

I discussed this with other co-workers on my project and we believe the lat_0=90 is also being used to set the scale factor somehow.  Per speculation but we're unable to arrive at another answer for why this is occurring.  We've looked over the GRIB documentation and have not found a place in the GRIB encoding where the map scale factor is specified.  Perhaps there's something we've overlooked. But as mentioned above, this does not occur for the GRIB files using other projections (actually it's only reported for Stereographic and Mercator but not Lambert Conformal).

The issue is minor because we did realize that setting k=1 solved the issue when using the proj4 string in Openlayers or Mapserver.  However, thought I should at least report this behavior for a possible fix.

## Steps to reproduce the problem.

Download any .bin file (these are the Alaska NDFD GRIB files) from 

ftp://tgftp.nws.noaa.gov/SL.us008001/ST.opnl/DF.gr2/DC.ndfd/AR.alaska/VP.001-003/

and perform gdalinfo on that downloaded file.

## Operating system

Not applicable.  Occurs in Linux and Windows.  However, used CentOS 6.10 to replicate the issue.

## GDAL version and provenance

2.2.4

**Comment by rouault** [comment]
This was corrected in GDAL 2.3.0

```
$ gdalinfo ds.maxrh.bin  --debug on
GNM: GNMRegisterAllInternal
GNM: RegisterGNMFile
GNM: RegisterGNMdatabase
OGRCT: Source: +proj=longlat +a=6371200 +b=6371200 +no_defs
OGRCT: Target: +proj=stere +lat_0=90 +lat_ts=60 +lon_0=210 +k=1 +x_0=0 +y_0=0 +a=6371200 +b=6371200 +units=m +no_defs
GDAL: GDALOpen(ds.maxrh.bin, this=0x1af6ea0) succeeds as GRIB.
Driver: GRIB/GRIdded Binary (.grb, .grb2)
GDAL: GDALDefaultOverviews::OverviewScan()
Files: ds.maxrh.bin
Size is 1649, 1105
Coordinate System is:
PROJCS["unnamed",
    GEOGCS["Coordinate System imported from GRIB file",
        DATUM["unknown",
            SPHEROID["Sphere",6371200,0]],
        PRIMEM["Greenwich",0],
        UNIT["degree",0.0174532925199433]],
    PROJECTION["Polar_Stereographic"],
    PARAMETER["latitude_of_origin",60],
    PARAMETER["central_meridian",210],
    PARAMETER["scale_factor",1],
    PARAMETER["false_easting",0],
    PARAMETER["false_northing",0]]
```

**Comment by teknocreator** [comment]
@rouault Thanks much for letting me know!  I will check about upgrading to 2.3.0 and confirm this.  Let you know if there's still an issue.
